python/tk_premiere/premiere.py

```diff
@@ -0,0 +1,263 @@
+
+import sgtk
+from sgtk.util.filesystem import ensure_folder_exists
+
+
+class ItemType(IntEnum):
+    """
+    Mapping for Adobe ProjectItem types.
+    """
+    CLIP = 1  # A clip
+    BIN = 2  # A bin
+    ROOT = 3  # Root of the project
+    FILE = 4  # A file
+
+
+class PremiereItem(object):
+    """
+    Base class for objects handling Premiere objects.
+    """
+    def __init__(self, item):
+        """
+        Instantiate a new :class:`PremiereItem` from the given item.
+
+        :param item: A Premiere object returned by the Adobe integration
+                              as a :class:`ProxyWrapper`
+        """
+        super(PremiereItem, self).__init__()
+        self._item = item
+
+    @property
+    def item(self):
+        """
+        Return the item associated with this instance.
+
+        :returns: A Premiere ProjectItem.
+        """
+        return self._item
+
+    @property
+    def name(self):
+        """
+        Return the name of this item.
+
+        :returns: A string.
+        """
+        return self._item.name
+
+    @property
+    def node_id(self):
+        """
+        Return the internal ID of this Premiere object.
+
+        :returns: A string.
+        """
+        return self._item.nodeId
+
+
+class PremiereProject(PremiereItem):
+    """
+    A class to handle Premiere projects.
+    """
+
+    @classmethod
+    def get_current_project(cls):
+        """
+        Return the current Premiere project, if any.
+
+        :returns: A :class:`PremiereProject`.
+        :raises ValueError: If no current project can be found.
+        """
+        engine = sgtk.platform.current_engine()
+        project = engine.adobe.app.project
+        if not project:
+            raise ValueError("No current Premiere project")
+
+        return cls(project)
+
+    def __init__(self, project):
+        """
+        Instantiate a new :class:`PremiereProject` for the given Premiere project.
+
+        :param project: A Premiere Project returned by the Adobe integration
+                        as a :class:`ProxyWrapper`.
+        """
+        super(PremiereProject, self).__init__(project)
+
+    @property
+    def path(self):
+        """
+        Return the path on the filesystem of this project.
+
+        :returns: A string.
+        """
+        return self._item.path
+
+    @property
+    def node_id(self):
+        """
+        Return the internal ID of the Premiere Project root item.
+
+        :returns: A string.
+        """
+        # The Project itself is not a Premiere ProjectItem but its
+        # root item is.
+        return self._item.rootItem.nodeId
+
+    @property
+    def bins(self):
+        """
+        Iterate over all bins in this project.
+
+        :yields: :class:`PremiereBin`.
+        """
+        # The top project item is also a bin.
+        bins = [self._item.rootItem]
+        while bins:
+            bin = bins.pop()
+            # Iterating over childre directly sometimes
```

Comment:
```suggestion
            # Iterating over children directly sometimes
```

Reply:
maybe it would be worth explaining what a bin is?

Reply:
Can't this method be generalized and put in `PremiereItem`, and depend on the type of node we're after? I don't know premiere enough to know...